lib/routes/account.js

```diff
@@ -1530,6 +1592,80 @@ module.exports = function (
     },
     {
       method: 'POST',
+      path: '/account/login/send_unblock_code',
+      config: {
+        validate: {
+          payload: {
+            email: validators.email().required()
+          }
+        }
+      },
+      handler: function (request, reply) {
+        log.begin('Account.SendUnblockCode', request)
+
+        var email = request.payload.email
+        var ip = request.app.clientAddress
+        var emailRecord
+
+        return customs.check(request, email, 'sendUnblockCode')
+          .then(lookupAccount)
+          .then(createUnblockCode)
+          .then(mailUnblockCode)
+          .done(function () {
+            reply({})
+          }, reply)
+
+        function lookupAccount() {
+          return db.emailRecord(email)
+            .then(function(record) {
+              emailRecord = record
+              return record.uid
+            })
+        }
+
+        function createUnblockCode(uid) {
+          return db.createUnblockCode(uid)
+        }
+
+        function mailUnblockCode(code) {
+          return getGeoData(ip)
+            .then(function (geoData) {
+              return mailer.sendUnblockCode(emailRecord, code, userAgent.call({
+                acceptLanguage: request.app.acceptLanguage,
+                ip: ip,
+                location: geoData.location,
+                timeZone: geoData.timeZone
+              }, request.headers['user-agent'], log))
+            })
+        }
+      }
+    },
+    {
+      method: 'POST',
+      path: '/account/login/reject_unblock_code',
+      config: {
+        validate: {
+          payload: {
+            uid: isA.string().max(32).regex(HEX_STRING).required(),
+            unblockCode: isA.string().regex(BASE_36).length(unblockCodeLen).required()
+          }
+        }
+      },
+      handler: function (request, reply) {
+        var uid = Buffer(request.payload.uid, 'hex')
+        var code = request.payload.unblockCode.toUpperCase()
+
+        log.begin('Account.RejectUnblockCode', request)
+        db.consumeUnblockCode(uid, code)
+          .then(
+            function () {
+              return {}
+            }
+          ).done(reply, reply)
+      }
+    },
+    {
```

Comment:
An endpoint that would be *extremely* useful for functional testing is one that causes a user's next signin to require unblock. We had similar functionality for account lockout so we could lock an account for end to end testing. That endpoint took the user's email address and password.